File: MATLAB/Quantize.m
function [x1, x2, x3] = Quantize(state, X1,X2,X3)
deltax1 = (X1(2)-X1(1))/2;
deltax2 = (X2(2)-X2(1))/2;
deltax3 = (X3(2)-X3(1))/2;

if state(1)>X1(end)+deltax1
    x1 = length(X1);
end

if state(2)>X2(end)+deltax2
    x2 = length(X2);
end

if state(3)>X3(end)+deltax3
    x3 = length(X3);
end

if state(1) <= X1(end) + deltax1
    x1 = find(abs(X1-state(1)) <= deltax1,1);
end

if state(2) <= X2(end) + deltax2
    x2 = find(abs(X2-state(2)) <= deltax2,1);
end

if state(3) <= X3(end) + deltax3
    x3 = find(abs(X3-state(3)) <= deltax3,1);
end

end

File: PythonBifurcated/Quantize.py
import numpy as np

INF = 100000


def Quantize(state, X1, X2):
    dist1 = INF
    dist2 = INF
    retval = np.zeros((1, 2))
    for x1 in X1:
        if np.abs(state[0, 0] - x1) < dist1:
            retval[0, 0] = x1
            dist1 = np.abs(state[0, 0] - x1)
    for x2 in X2:
        if np.abs(state[0, 1] - x2) < dist2:
            retval[0, 1] = x2
            dist2 = np.abs(state[0, 1] - x2)
    return retval
